pulser-core/pulser/sequence/sequence.py

```diff
@@ -370,6 +398,7 @@ def config_slm_mask(self, qubits: Iterable[QubitId]) -> None:
         # If checks have passed, set the SLM mask targets
         self._slm_mask_targets = targets
 
+    @seq_decorators.screen
     def switch_device(
```

Comment:
What is the advantage of this method over rebuilding the same sequence with a different device? Or is it just for convenience?

Reply:
It's just a convenience. In particular, it can make sure for you that the sequence remains the same on the new device, which can be useful.